builtin/providers/aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -91,49 +188,168 @@ func resourceAwsKinesisFirehoseDeliveryStream() *schema.Resource {
 	}
 }
 
+func validateConfiguration(d *schema.ResourceData) error {
+	destination := d.Get("destination").(string)
+	if destination != "s3" && destination != "redshift" {
+		return fmt.Errorf("[ERROR] Destination must be s3 or redshift")
+	}
+
+	return nil
+}
+
+func createS3Config(d *schema.ResourceData) *firehose.S3DestinationConfiguration {
+	s3 := d.Get("s3_configuration").([]interface{})[0].(map[string]interface{})
```

Comment:
Just noting the types are guaranteed by helper/schema and the length is guaranteed by Required: true + MaxItems: 1